draft-irtf-cfrg-aegis-aead.md

```diff
@@ -1539,23 +1562,27 @@ AEGIS-256 offers 256-bit message security against plaintext and state recovery,
+Protocols mandating a fully committing scheme without that restriction can provide the associated data as input to a cryptographic hash function and use the output as the `ad` parameter of the `Encrypt` and `Decrypt` functions. The selected hash function must ensure a minimum of 128-bit preimage resistance. An instance of such a function is SHA-256 {{!RFC6234}}.
 
-Under the assumption that the secret key is unknown to the attacker both AEGIS-128L and AEGIS-256 target 128-bit security against forgery attacks regardless of the tag size.
+Under the assumption that the secret key is unknown to the attacker, all AEGIS variants target 128-bit security against forgery attacks regardless of the tag size.
 
-Both algorithms MUST be used in a nonce-respecting setting: for a given `key`, a `nonce` MUST only be used once. Failure to do so would immediately reveal the bitwise difference between two messages.
+AEGIS algorithms MUST be used in a nonce-respecting setting: for a given `key`, a `nonce` MUST only be used once. Failure to do so would immediately reveal the bitwise difference between two messages.
 
-If tag verification fails, the decrypted message and wrong message authentication tag MUST NOT be given as output. As shown in {{VV18}}, even a partial leak of the plaintext without verification would facilitate chosen ciphertext attacks.
+If tag verification fails, the unverified plaintext and the computed message authentication tag MUST NOT be released. As shown in {{VV18}}, even a partial leak of the plaintext without verification would facilitate chosen ciphertext attacks.
 
 Every key MUST be randomly chosen from a uniform distribution.
 
 The nonce MAY be public or predictable. It can be a counter, the output of a permutation, or a generator with a long period.
 
-With AEGIS-128L, random nonces can safely encrypt up to 2<sup>48</sup> messages using the same key with negligible (~ 2<sup>-33</sup>, to align with NIST guidelines) collision probability.
+With AEGIS-128L and AEGIS-128X, random nonces can safely encrypt up to 2<sup>48</sup> messages using the same key with negligible (~ 2<sup>-33</sup>, to align with NIST guidelines) collision probability.
+
+With AEGIS-256 and AEGIS-256X, random nonces can be used with no practical limits.
 
-With AEGIS-256, random nonces can be used with no practical limits.
+AEGIS nonces match the size of the key. AEGIS-128L and AEGIS-128X feature 128-bit nonces, offering an extra 32 bits compared to the commonly used AEADs in IETF protocols. The AEGIS-256 and AEGIS-256X variants provide an even ampler space for nonces, surpassing the 192-bit requirement for secure utilization of random nonces without practical limitations.
+
+In all these variants, unused nonce bits can encode a key identifier, enhancing multi-user security. If every key has a unique identifier, multi-target attacks don't provide any advantage over single-target attacks.
```

Comment:
Is it worth saying something like `There are two ways to mitigate multi-target attacks, namely use an AEGIS variant with a 256-bit key and/or encode a key identifier in the nonce`. And then perhaps an example of a key identifier (e.g. random bits)? Otherwise, I like what you've put.

Reply:
Maybe there's some confusion here.

A key identifier can be included both in 128-bit and 256-bit variants.

In 128-bit variants, 96-bit can be used like other AEADs, leaving 32 bits to encode a key ID.
In 256-bit variants, there's plenty of room so that even with 192-bit nonces, there's still 96 bits left to encode a key ID.

The mitigation is the same in both cases.

Reply:
The format of a key identifier is really application-specific. It can be a used ID. It can be a SQL row index. It can be a timestamp. So I'd rather not enter that territory. 

It would be also good to keep that section short. Multi-user security is good, but with a large key space, not a practical concern. We shouldn't put too much emphasis on it; people may think that AEGIS requires a peculiar nonce format to be secure.

Reply:
What I meant was instead of doing a key identifier, you can use a 256-bit key. Just thinking for completeness really.

Fair enough about the format, and you're right.

Reply:
With repeated nonces, even with a 256-bit key, security will degrade with the number of targeted keys. With a very large key space, an exhaustive key search is indeed unlikely to be successful no matter what the number of keys are targeted.

But multi-user security as defined in the other draft specifically refers to no degradation or graceful degradation. So I'm not sure that using 256-bit key can be mentioned as an alternative here.

Reply:
Ok that makes sense. Happy to leave it then.